### PR TITLE
match expressions documenting

### PR DESCRIPTION
**Pull request by SilentCicero**
- Document exhaustive match expressions

Closes: https://github.com/FuelLabs/sway/issues/1263

### CODE REVIEW

File: docs/src/basics/control_flow.md [hunk 1]
```diff
@@ -34,6 +34,28 @@ let my_data = if some_bool < 10 { foo() } else { bar() };
 
 Note that all branches of the `if` expression must return a value of the same type.
 
+### `match` expressions
+
+Sway supports advanced pattern matching through exhaustive _match_ expressions.
+
+```sway
+let x = 5;
```
Comment (adlerjohn) [inline docs/src/basics/control_flow.md:42]: Good, but put this in the `examples` directory.

File: docs/src/basics/control_flow.md [hunk 2]
```diff
@@ -34,6 +34,28 @@ let my_data = if some_bool < 10 { foo() } else { bar() };
 
 Note that all branches of the `if` expression must return a value of the same type.
 
+### `match` expressions
+
+Sway supports advanced pattern matching through exhaustive _match_ expressions.
```
Comment (adlerjohn) [inline docs/src/basics/control_flow.md:39]:
```suggestion
Sway supports advanced pattern matching through exhaustive `match` expressions.
```

File: docs/src/basics/control_flow.md [hunk 3]
```diff
@@ -34,6 +34,28 @@ let my_data = if some_bool < 10 { foo() } else { bar() };
...
+
+```sway
+let x = 5;
+let a = match 8 {
+    7 => { 4 },
+    9 => { 5 },
+    8 => { 6 },
+    _ => { 100 },
+};
+let b = match x {
```
Comment (adlerjohn) [inline docs/src/basics/control_flow.md:49]: There should be comments to illustrate both match as an expression and match as a statement for control flow.

**Review by adlerjohn — CHANGES_REQUESTED** [review]

**Comment by mohammadfawaz** [comment]
Thanks! If we're planning on covering `if let` in a separate PR, then we should make sure that https://github.com/FuelLabs/sway/issues/1263 is not closed once this PR is merged.

**Comment by SilentCicero** [comment]
> Thanks! If we're planning on covering `if let` in a separate PR, then we should make sure that #1263 is not closed once this PR is merged.

I think If let is already documented no?

**Comment by SilentCicero** [comment]
Abandoning for now.

**Pull request closed without merging**